app/policies/application_policy.rb

```diff
@@ -3,6 +3,47 @@ class ApplicationPolicy
 
   def initialize(user, record)
     @user = user
-    @record = record.to_model
+    @record = record
```

Comment:
Could you please use [ActionPolicy](https://github.com/palkan/action_policy) gem for authorization instead of Pundit?
ActionPolicy is faster and more modern.

Documentation: https://actionpolicy.evilmartians.io/#/